### PR TITLE
Added Biodiesel to the game

### PR DESCRIPTION
**Pull request by ghost**
Biodiesel is a kind of renewable fuel resource made of (m)ethanol and animal lipids.

I added 'ethanol', 'methanol', 'biodiesel', 'i6 biodiesel engine', 'biodiesel kiln' and 'Biodiesel textbook' to the game. The book 'Biodiesel: Renewable Fuel Resource' contains recipes to craft biodiesel engines and biodiesel kilns. 

After following the steps below, we get our renewable resource powered vehicle.
0) Search for the book 'Biodiesel: Renewable Fuel Resource' and read it to learn recipes.
1) Remove the original petro-diesel engine from our vehicle, prepare needed components and craft the Biodiesel Engine, then install it back to our vehicle.
2) Prepare a large amount of alcohol(eg. Vodka, Bum Wine), get ethanol and methanol after fractionation.
3) Prepare needed components to craft the empty biodiesel kiln.
4) Prepare enough animal lipids(eg. Lard, Tainted Tallow), ethanol or methanol and battery(the chemical reaction is heated by electric power) to craft filled biodiesel kiln.
5) Activate the biodiesel kiln and wait for 6 hours, then biodiesel is ready to harvest.

### CODE REVIEW

**Comment by Coolthulhu** [comment]
The idea is good, our existing diesel recipe is pretty much technomagic and this looks closer to actual fuel manufacturing process.

But:

Having a separate set of diesel engines just for biodiesel is unnecessary. It could just produce the regular diesel. Biodiesel may have negative effects on unmodified engine IRL, but this would be a minor issue, while requiring copying every diesel engine (and adding a new recipe) is a major one.
It can be handwaved as the survivor modifying the engine when making the first batch of biodiesel, as it is unlikely that anyone will produce biodiesel before getting a complete vehicle.

Kilns are unwieldy and tedious to use. They don't allow batch crafting and require way too many keypresses. It would be better off as a regular recipe, without the kilning step. Or with the kiln as a tool.
Also, not having kilns would allow using car batteries to fuel the reaction, rather than having to use the itemized batteries.

Balancing: "weak" alcohol is only 1/2 as strong as medium alcohol and 1/3 as strong alcohol. Bum wine would only yield 1/3 of the ethanol vodka would.

Alcohols: it looks like the methanol recipe will produce just as much methanol as ethanol. Food-grade alcohol contains only minimal amount of methanol. For now it would be better off without methanol at all, it can be added when needed and when there's a good source for it. Also, many alcohols are missing from the list, though that's a minor issue that can easily be fixed later.

Finally; code quality. You have tabs in few places. We use 4 spaces for indentation.

**Comment by DLaboratory** [comment]
Thank you for replying. But I think I really have to add a new set of engine and fuel in order to make biodiesel powered vehicles weaker than petro-diesel powered ones and have larger possibility of engine backfire. How about adding a 'Biodiesel Reacting Box' like fermenting vat instead of kiln? As long as the player throws tainted tallow and ethanol into that furniture(and need electric power, charge from dry cells or car batteries in a menu), the reaction will start at the moment.

**Comment by DLaboratory** [comment]
Now I removed biodiesel kilns and wrote a recipe for crafting biodiesel in the & menu.

File: data/json/items/var_vehicle_parts.json [hunk 1]
```diff
@@ -416,6 +416,25 @@
         "category": "veh_parts",
         "bigness-aspect": "ENGINE_DISPLACEMENT"
     },
+    {
+        "type":"VAR_VEH_PART",
+        "id": "i6_biodiesel",
+        "symbol": ":",
+        "color": "light_cyan",
+        "name": "I6 biodiesel engine",
+        "description": "A straight-6 biodiesel engine, a little weaker than petro fuel engines because it drinks biodiesel.",
```
Comment (Darkyhard) [inline data/json/items/var_vehicle_parts.json]: Typo here: "petro" instead of "petrol".

Comment (DLaboratory) [inline data/json/items/var_vehicle_parts.json]: @Darkyhard Thank you!

File: data/json/item_groups.json [hunk 2]
```diff
@@ -2015,6 +2015,7 @@
       ["foot_crank", 10],
       ["cable", 60],
       ["textbook_mechanics", 12],
+	  ["textbook_biodiesel", 10],
```
Comment (Coolthulhu) [inline data/json/item_groups.json]: Those are still tabs

**Comment by kevingranade** [comment]
Why do biodiesel engines need to be weaker than petrol engines?
As Coolthulu says, if this gets added in this state, people will (justifiably) try to add biodiesel versions of every diesel engine, cluttering up the json.  A quick survey of sources indicates that biodiesel is about 8% less energy dense than regular diesel, but since the limiting factor for a survivor is going to be total supply rather than vehicle capacity, it doesn't make much sense to change the engine's performance, also I'm not seeing any numbers on changed HP for switching to biodiesel, just fuel economy.  I suspect once you tune flow rates the HP for the engine is the same before or after switching to biodiesel.
More importantly, treating biodiesel interchangeably with diesel lets players refuel with either in the same engine, which is one of the major selling points of biodiesel conversion in the first place.

**Pull request closed without merging**